sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -269,6 +274,10 @@ static int sof_rt5682_hw_params(struct snd_pcm_substream *substream,
 			clk_freq = 24000000;
 		else
 			clk_freq = 19200000;
+
+		/* use the tplg configured mclk if existed */
+		if (dai && dai->dai_config && dai->dai_config->ssp.mclk_rate)
+			clk_freq = dai->dai_config->ssp.mclk_rate;
```

Comment:
It may make more sense here to have an API call for this e.g. ```snd_sof_dai_get_ssp_mclk``` as this may be needed in other machines too. It can be exported by the Intel SOF code.

Reply:
> It may make more sense here to have an API call for this e.g. `snd_sof_dai_get_ssp_mclk` as this may be needed in other machines too. It can be exported by the Intel SOF code.

yes, let me think of doing it like that.